db/python/tables/output_file.py

```diff
@@ -121,7 +121,7 @@ async def add_output_file_to_analysis(
         """Add file to an analysis (through the join table)"""
         _query = dedent(
             """
-            INSERT INTO analysis_outputs
+            INSERT IGNORE INTO analysis_outputs
```

Comment:
It feels like a code smell :-)
wonder if we should report the error IGNORE is trying to hide or calling code should be revisited.
Is this purely to prevent duplicates?

Reply:
Yep, purely to prevent duplicates, I could replace this with an `ON DUPLICATE` clause, but I don't quite understand/know if one is better or preferred over the other. Thoughts?

Reply:
There is a discussion regarding this:
https://stackoverflow.com/questions/548541/insert-ignore-vs-insert-on-duplicate-key-update

Someone mentioned ON DUPLICATE can cause mysql deadlocking.
I do not have preference/strong opinion on this as both are non standard SQL.
Let's stick with IGNORE as you had implemented. We can alway revisit if needed.